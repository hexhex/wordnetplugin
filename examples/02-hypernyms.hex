%% this returns all hypernyms of a word
%% in all parts of speech

%% the search number for hypernyms is 2

word("house").
hypernyms(P,W,SW,H,SH) :- &wordnet[2,W](P,SW,H,SH), word(W).

hypernyms(1,"house",1,"building",2), 
hypernyms(1,"house",1,"home",1), 
hypernyms(1,"house",2,"concern",1), 
hypernyms(1,"house",6,"folk",1), 
hypernyms(1,"house",6,"family",1), 
hypernyms(2,"house",2,"shelter",1)?
