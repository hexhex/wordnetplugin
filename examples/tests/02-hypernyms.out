{hypernyms(1,"house",1,"edifice",2), hypernyms(1,"house",1,"building",2), hypernyms(1,"house",1,"dwelling_house",1), hypernyms(1,"house",1,"habitation",1), hypernyms(1,"house",1,"abode",1), hypernyms(1,"house",1,"domicile",1), hypernyms(1,"house",1,"home",1), hypernyms(1,"house",1,"dwelling",1), hypernyms(1,"house",2,"business_organisation",1), hypernyms(1,"house",2,"business_organization",1), hypernyms(1,"house",2,"business_concern",1), hypernyms(1,"house",2,"concern",1), hypernyms(1,"house",2,"business",1), hypernyms(1,"house",3,"community",1), hypernyms(1,"house",4,"audience",1), hypernyms(1,"house",5,"law-makers",1), hypernyms(1,"house",5,"general_assembly",1), hypernyms(1,"house",5,"legislative_body",1), hypernyms(1,"house",5,"legislative_assembly",1), hypernyms(1,"house",5,"legislature",1), hypernyms(1,"house",6,"phratry",1), hypernyms(1,"house",6,"sept",1), hypernyms(1,"house",6,"kinsfolk",1), hypernyms(1,"house",6,"kinfolk",1), hypernyms(1,"house",6,"folk",1), hypernyms(1,"house",6,"family_line",1), hypernyms(1,"house",6,"family",1), hypernyms(1,"house",7,"child's_play",1), hypernyms(1,"house",7,"play",1), hypernyms(1,"house",8,"part",1), hypernyms(1,"house",8,"region",1), hypernyms(1,"house",9,"management",1), hypernyms(1,"house",10,"social_unit",1), hypernyms(1,"house",10,"unit",1), hypernyms(1,"house",11,"edifice",1), hypernyms(1,"house",11,"building",1), hypernyms(1,"house",12,"edifice",1), hypernyms(1,"house",12,"building",1), hypernyms(2,"house",1,"admit",1), hypernyms(2,"house",1,"hold",1), hypernyms(2,"house",1,"accommodate",1), hypernyms(2,"house",2,"shelter",1), word("house")}
